### PR TITLE
Fixes `TypeError: Cannot read property 'id' of undefined`

### PR DESCRIPTION
**Pull request by u9g**
Fixes: 
```
C:\Users\---\Documents\js-projects\cosmic-lore-bot\node_modules\mineflayer\lib\plugins\inventory.js:444
    while (actionId > click.id) {
                            ^

TypeError: Cannot read property 'id' of undefined
    at Client.<anonymous> (C:\Users\---\Documents\js-projects\cosmic-lore-bot\node_modules\mineflayer\lib\plugins\inventory.js:576:5)
    at Client.emit (node:events:394:28)
    at FullPacketParser.<anonymous> (C:\Users\---\Documents\js-projects\cosmic-lore-bot\node_modules\minecraft-protocol\src\client.js:91:12)
    at FullPacketParser.emit (node:events:394:28)
    at addChunk (C:\Users\---\Documents\js-projects\cosmic-lore-bot\node_modules\readable-stream\lib\_stream_readable.js:298:12)
    at readableAddChunk (C:\Users\---\Documents\js-projects\cosmic-lore-bot\node_modules\readable-stream\lib\_stream_readable.js:280:11)
    at FullPacketParser.Readable.push (C:\Users\---\Documents\js-projects\cosmic-lore-bot\node_modules\readable-stream\lib\_stream_readable.js:241:10)
    at FullPacketParser.Transform.push (C:\Users\---\Documents\js-projects\cosmic-lore-bot\node_modules\readable-stream\lib\_stream_transform.js:139:32)
    at FullPacketParser._transform (C:\Users\---\Documents\js-projects\cosmic-lore-bot\node_modules\protodef\src\serializer.js:89:10)
```
closes https://github.com/PrismarineJS/mineflayer/issues/1811

### CODE REVIEW

**Comment by U5B** [comment]
Fix makes ci fail, seemingly for all the tests involving inventory... (except 1.17 since it doesn't use the transaction packet)

**Comment by u9g** [comment]
> Fix makes ci fail, seemingly for all the tests involving inventory... (except 1.17 since it doesn't use the transaction packet)

Well that doesn't seem very good

**Comment by U5B** [comment]
Do we need a do loop instead?

File: lib/plugins/inventory.js [hunk 1]
```diff
@@ -434,6 +434,13 @@ function inject (bot, { version, hideErrors }) {
       }
       return
     }
+    if (windowClickQueue.length > 0 && Math.max(...windowClickQueue.map(x => x.id)) < actionId) {
+      if (!hideErrors) {
+        console.log('WARNING : ignoring transaction where actionId > max action id in windowClickQueue and dropping all of windowClickQueue')
+        windowClickQueue.pop()
```
Comment (U5B) [inline lib/plugins/inventory.js:440]: move `windowClickQueue.pop()` out of `if (!hideErrors)`

Comment (u9g) [reply lib/plugins/inventory.js]: done :)

**Comment by rom1504** [comment]
Can you rebase on master?

**Comment by u9g** [comment]
closing, pr #2217 supersedes

**Pull request closed without merging**